changedetectionio/model/Watch.py

```diff
@@ -151,6 +153,43 @@ def save_history_text(self, contents, timestamp):
         #@todo bump static cache of the last timestamp so we dont need to examine the file to set a proper ''viewed'' status
         return snapshot_fname
 
+    # Save previous text snapshot for diffing
+    def save_previous_text(self, contents):
+        from os import mkdir
+        import logging
+
+        output_path = "{}/{}".format(self.__datastore_path, self['uuid'])
+
+        # Incase the operator deleted it, check and create.
+        if not os.path.isdir(output_path):
+            mkdir(output_path)
+
+        snapshot_fname = "{}/previous.txt".format(output_path)
+        logging.debug("Saving previous text {}".format(snapshot_fname))
+
+        with open(snapshot_fname, 'wb') as f:
+            f.write(contents)
+            f.close()
+
+        return snapshot_fname
+
+    # Get previous text snapshot for diffing
+    def get_previous_text(self):
+        from os import path
+
+        output_path = "{}/{}".format(self.__datastore_path, self['uuid'])
+
+        snapshot_fname = "{}/previous.txt".format(output_path)
+        if not path.isfile(snapshot_fname):
```

Comment:
Please don't do this kind of 'mystery logic' where it does something that's not written in the function name, it's best to fix it so that this situation never happens (like check `watch.history_n > 2`)

Reply:
Ok I will change it to check the history length first and if it does not exist just return an empty string

Reply:
better to let it throw an exception i think, that case should never happen

Reply:
If I throw an exception, it will always throw the exception since we are not saving the previous text until after it determines if a change was done or not. Going to keep it to returning an empty string but only `if self.history_n < 1` . It will no longer create the file if it does not exist